### PR TITLE
test: Tests and dedicated directory for the SupersetResourceSelect Component

### PR DESCRIPTION
**Pull request by geido**
Moved the SupersetResourceSelect Component to its dedicated directory while refactoring the tests with react-testing-library.

### CODE REVIEW

File: superset-frontend/src/components/SupersetResourceSelect/SupersetResourceSelect.test.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  const select = screen.getByText('Select...');
+  userEvent.click(select);
+  expect(screen.getByText('Loading...')).toBeInTheDocument();
+});
+
+test('should render the No options message', async () => {
+  render(<SupersetResourceSelect {...mockedProps} />);
+  const select = screen.getByText('Select...');
+  userEvent.click(select);
+  await waitFor(() => {
```
Comment (michael-s-molina) [inline superset-frontend/src/components/SupersetResourceSelect/SupersetResourceSelect.test.tsx:55]: Nit. Here you can use `findBy`.

File: superset-frontend/src/components/SupersetResourceSelect/SupersetResourceSelect.test.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    expect(screen.getByText('No options')).toBeInTheDocument();
+  });
+});
+
+test('should render the typed text', async () => {
+  render(<SupersetResourceSelect {...mockedProps} />);
+  const select = screen.getByText('Select...');
+  userEvent.click(select);
+  userEvent.type(select, 'typed text');
+  expect(screen.getByText('typed text')).toBeInTheDocument();
```
Comment (michael-s-molina) [inline superset-frontend/src/components/SupersetResourceSelect/SupersetResourceSelect.test.tsx:65]: Shouldn't you await with `findBy` here?

**Review by michael-s-molina — APPROVED** [review]
LGTM

**Comment by geido** [comment]
@junlincc @rusackas

**Review by pkdotson — APPROVED** [review]
LGTM!

**Pull request merged**